pkg/processor/worker/worker.go

```diff
@@ -151,20 +151,25 @@ func (w *Worker) SupportsRestart() bool {
 	return w.runtime.SupportsRestart()
 }
 
-func (w *Worker) Drain() error {
-	err := w.runtime.Drain()
-	if err == nil {
+func (w *Worker) IsDrained() bool {
+	w.drainedLock.Lock()
+	defer w.drainedLock.Unlock()
 
-		w.setDrained(true)
-	}
-	return err
+	return w.isDrained
 }
 
-func (w *Worker) IsDrained() bool {
+func (w *Worker) Drain() error {
 	w.drainedLock.Lock()
 	defer w.drainedLock.Unlock()
 
-	return w.isDrained
+	if !w.isDrained {
```

Comment:
Use `IsDrained()` function that already uses the lock

Reply:
OK I see that now.
So then - is `IsDrained` being used anywhere at all now?
If not you can remove it.

Reply:
@TomerShor nice catch, it was only one place we used it. Deleted :)